### PR TITLE
avm1: tellTarget on an undefined object targets the base clip in SWF6-

### LINKED ISSUE #13157: Regression in Curveball: Clicking on Start Game or High Scores unresponsive
### Describe the bug

I find both on desktop and web that as of nightly 8/29/23, the buttons to either start the game or view high scores are unresponsive when clicked.

### Expected behavior

Buttons should be clickable
[curveball.zip](https://github.com/ruffle-rs/ruffle/files/12583277/curveball.zip)

### Affected platform

Online demo

### Operating system

Windows 11

### Browser

Chromium Edge

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by n0samu**
This restores the old behavior from before #12956, only for SWF6 and below. It turns out this behavior is version-dependent.

Fixes #13157.

### CODE REVIEW

**Review by sleepycatcoding — APPROVED** [review]

File: core/src/avm1/activation.rs [hunk 1]
```diff
@@ -1981,7 +1981,12 @@ impl<'a, 'gc> Activation<'a, 'gc> {
                 return self.set_target(&target);
             }
             Value::Undefined => {
-                self.set_target_clip(None);
+                // In SWF6 and below, SetTarget2 on an undefined object resets the target to the base clip
+                if self.swf_version() > 6 {
```
Comment (Lord-McSweeney) [inline core/src/avm1/activation.rs:1985]: Weird, the SWFs that were fixed by https://github.com/ruffle-rs/ruffle/pull/12956 were both SWFv6. Have you tested those SWFs with this PR?

Comment (sombraguerrero) [reply core/src/avm1/activation.rs:1985]: I just did and you're right. Black 2 goes back to locking up.
[308750_BLACK.zip](https://github.com/ruffle-rs/ruffle/files/12701985/308750_BLACK.zip)
Here's the actual game as opposed to the NG launcher.

**Comment by Lord-McSweeney** [comment]
After further investigation, nosamu found that the behavior differs in SWFv4/5, SWFv6, and SWFv7+.

File: core/src/avm1/activation.rs (hunk 1 above)
Comment (sombraguerrero) [reply core/src/avm1/activation.rs:1985]: I retract part of my comment this morning. This PR does work against both Curveball and Black 2 as expected. I did not properly clone the branch when I tested earlier today :P 
I advocate for this to be approved and merged.

Comment (n0samu) [reply core/src/avm1/activation.rs:1985]: Actually both games are SWFv7 and should not be affected by this PR. @sombraguerrero Please test both games and let us know what you find. This PR should be fine but please confirm.

Comment (sombraguerrero) [reply core/src/avm1/activation.rs:1985]: Yes, confirmed. Both Black 1 and 2 are working with the PR and JPEXs reads them both as 7.

**Review by adrian17 — APPROVED** [review]

**Pull request merged**